pkg/arvo/sys/vane/clay.hoon

```diff
@@ -4199,6 +4472,243 @@
       --
     --
   --
+::  userspace agent management
+::
+++  lu
+  |=  [now=@da rof=roof hen=duct raft]
+  =*  ruf  |3.+<.$
+  =|  mow=(list move)
+  |%
+  ++  abet
+    ^-  [(list move) raft]
+    [(flop mow) ruf]
+  ::
+  ++  emit
+    |=  mof=move
+    %_(+> mow [mof mow])
+  ::
+  ++  emil
+    |=  mof=(list move)
+    %_(+> mow (weld (flop mof) mow))
+  ::  +ford: init ford
+  ::
+  ++  ford
+    |=  [her=ship syd=desk yon=(unit aeon)]
+    =/  den  ((de now rof hen ruf) her syd)
+    (aeon-ford:den ?~(yon let.dom:den u.yon))
+  ::  +wrap: save ford cache
+  ::
+  ++  wrap
+    |*  [her=ship syd=desk yon=(unit aeon) res=* =state:ford:fusion]
+    =^  moves  ruf
+      =/  den  ((de now rof hen ruf) her syd)
+      abet:+:(aeon-flow:den ?~(yon let.dom:den u.yon) res cache.state &2.state)
+    [res (emil moves)]
+  ::
+  ++  trace
+    |=  [pri=@ print=(trap tape)]
+    ?:  (lth veb.bug pri)
+      same
+    (slog leaf+"goad: {(print)}" ~)
+  ::  +goad: emit %load move for all desks, applying $rein's
+  ::
+  ::  [goad] Must be called any time the set of running agents changes.
+  ::  This is whenever an agent is started, stopped, or updated.
+  ::
+  ::  This is not move-order agnostic -- you must be careful of
+  ::  reentrancy as long as arvo's move order is depth-first.
+  ::
+  ::  [tare] >
+  ::
+  ++  goad
+    ^+  ..abet
+    =^  sat=(list [=desk =bill])  ..abet
+      =/  desks=(list desk)  ~(tap in ~(key by dos.rom))
+      |-  ^-  [(list [desk bill]) _..abet]
+      ?~  desks
+        [~ ..abet]
+      =/  den  ((de now rof hen ruf) our i.desks)
+      ?.  =(%live liv.dom.den)
+        %-  (trace 2 |.("{<i.desks>} is not live"))
+        $(desks t.desks)
+      =^  res  den  (aver:den ~ %x da+now /desk/bill)
+      =.  ruf  +:abet:den
+      ?.  ?=([~ ~ *] res)
+        $(desks t.desks)
+      =/  bill  ~|  [%building-bill i.desks]  !<(bill q.u.u.res)
+      =/  rid  (override bill ren.dom.den)
+      %-  %+  trace  2  |.
+          "{<i.desks>} has bill {<bill>} and rein {<ren.dom.den>}, so {<rid>}"
+      =^  sats  ..abet  $(desks t.desks)
+      [[[i.desks rid] sats] ..abet]
+    ::
+    =.  sat  (apply-precedence sat)
+    =+  ?:  (lth veb.bug 1)  ~
+        %.  ~  %-  slog
+        %+  turn  sat
+        |=  [=desk =bill]
+        leaf+"goad: output: {<desk>}: {<bill>}"
+    =^  agents  ..abet  (build-agents sat)
+    =.  ..abet
+      (build-marks (turn (skip sat |=([desk =bill] =(bill ~))) head))
```

Comment:
![Screenshot from 2022-11-21 12-02-14](https://user-images.githubusercontent.com/5553176/203137968-44459918-cc7b-46af-8c85-5ed7e0f483ba.png)

Reply:
It is certainly at any rate a reasonable request to know *whether* all your mark files compile, to not have any files of marks that don't, to eagerly abort any computation touching non-file values of marks that don't compile, etc